src/main/scala/mesosphere/marathon/MarathonScheduler.scala

```diff
@@ -187,18 +188,31 @@ class MarathonScheduler @Inject() (
 
   override def error(driver: SchedulerDriver, message: String) {
     log.warn("Error: %s".format(message))
-    suicide()
+
+    // Currently, it's pretty hard to disambiguate this error from other causes of framework errors.
+    // Watch MESOS-2522 which will add a reason field for framework errors to help with this.
+    // For now the frameworkId is removed for all messages.
+    val removeFrameworkId = true
+    suicide(removeFrameworkId)
   }
 
-  private def suicide(): Unit = {
-    log.fatal("Committing suicide")
+  /**
+    * Exit the JVM process.
+    * Optionally deleting the frameworkId. Depending on the cause of this error,
```

Comment:
Needs rewording for clarity IMO.

How about:

```
/**
  * Exits the JVM process, optionally deleting Marathon's FrameworkID
  * from the backing persistence store.
  *
  * If `removeFrameworkId` is set, the next Marathon process elected
  * leader will fail to find a stored FrameworkID and invoke `register`
  * instead of `reregister`.  This is important because on certain kinds
  * of framework errors (such as exceeding the framework failover timeout),
  * the scheduler may never re-register with the saved FrameworkID until
  * the leading Mesos master process is killed.
  */